flink-python/pyflink/table/catalog.py

```diff
@@ -0,0 +1,965 @@
+        """
+        Get a deep copy of the CatalogPartition instance.
+
+        :return: A copy of CatalogPartition instance.
+        """
+        return CatalogPartition(self._j_catalog_partition.copy())
+
+    def get_description(self):
+        """
+        Get a brief description of the partition object.
+
+        :return: An optional short description of partition object.
+        """
+        description = self._j_catalog_partition.getDescription()
+        if description.isPresent():
+            return description.get()
+        else:
+            return None
+
+    def get_detailed_description(self):
+        """
+        Get a detailed description of the partition object.
+
+        :return: An optional long description of the partition object.
+        """
+        detailed_description = self._j_catalog_partition.getDetailedDescription()
+        if detailed_description.isPresent():
+            return detailed_description.get()
+        else:
+            return None
+
+
+class CatalogFunction(object):
+    """
+    Interface for a function in a catalog.
+    """
+
+    def __init__(self, j_catalog_function):
+        self._j_catalog_function = j_catalog_function
+
+    @staticmethod
+    def _get(j_catalog_function):
+        if j_catalog_function.getClass().getName() == \
+                "org.apache.flink.table.catalog.hive.HiveCatalogFunction":
+            return HiveCatalogFunction(j_hive_catalog_function=j_catalog_function)
+        else:
+            return CatalogFunction(j_catalog_function)
+
+    def get_class_name(self):
+        """
+        Get the full name of the class backing the function.
+
+        :return: The full name of the class.
+        """
+        return self._j_catalog_function.getClassName()
+
+    def get_properties(self):
+        """
+        Get the properties of the function.
+
+        :return: The properties of the function.
+        """
+        return dict(self._j_catalog_function.getProperties())
+
+    def copy(self):
+        """
+        Create a deep copy of the function.
+
+        :return: A deep copy of "this" instance.
+        """
+        return CatalogFunction(self._j_catalog_function.copy())
+
+    def get_description(self):
+        """
+        Get a brief description of the function.
+
+        :return: An optional short description of function.
+        """
+        description = self._j_catalog_function.getDescription()
+        if description.isPresent():
+            return description.get()
+        else:
+            return None
+
+    def get_detailed_description(self):
+        """
+        Get a detailed description of the function.
+
+        :return: An optional long description of the function.
+        """
+        detailed_description = self._j_catalog_function.getDetailedDescription()
+        if detailed_description.isPresent():
+            return detailed_description.get()
+        else:
+            return None
+
+
+class ObjectPath(object):
```

Comment:
I think It's better to add `equals` and `hashCode` for complete align JAVA. What do you think?